jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClientManager.java

```diff
@@ -98,25 +98,13 @@ public MucClientManager()
      * @param features the features to use for disco#info.
      */
     public MucClientManager(String[] features)
-    {
-        this(features,
-             ExecutorUtils.newCachedThreadPool(
-                 true,
-                 MucClientManager.class.getSimpleName()));
-    }
-
-    /**
-     * Initializes a new {@link MucClientManager} instance.
-     *
-     * @param features the features to use for disco#info.
-     */
-    public MucClientManager(String[] features, Executor executor)
     {
         SmackConfiguration.setUnknownIqRequestReplyMode(
             SmackConfiguration.UnknownIqRequestReplyMode
                 .replyFeatureNotImplemented);
 
-        this.executor = executor;
+        this.executor = ExecutorUtils.newScheduledThreadPool(
+            3, true, MucClientManager.class.getSimpleName());
```

Comment:
Does this switch to a fixed-size thread pool which will execute the the XMPP connect/login? This will not work well when the number of MucClients is large

Reply:
Yes, it does. You are worried that it will not connect quickly?

Reply:
it will connect at most 3 clients at a time, what else do you propose?